### PR TITLE
Feature/cc 0046 add husky

### PR DESCRIPTION
**Pull request by GitHubEason**
Add husky into this repo， Husky will run prettier and test， before commit.

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -1,4 +1,6 @@
 {
   "singleQuote": true,
-  "trailingComma": "all"
-}
\ No newline at end of file
+  "trailingComma": "all",
+  "tabWidth": 2,
+  "printWidth": 100
+}
```
Comment (tatoMa) [inline .prettierrc:6]: Do you think we might need this?
"endOfLine": "auto"
The default value of endOfLine is "lf" based on the document. It is mainly for Linux and Macos users.

https://prettier.io/docs/en/options.html#end-of-line

**Review by tatoMa — COMMENTED** [review]
Do you think we also need to add lint in pre-commit?

File: .prettierrc (hunk 1 above)
Comment (GitHubEason) [reply .prettierrc:6]: yep, thanks for reminding me,  I will add it soon

Comment (Jzeva) [reply .prettierrc:6]: And prettier please. We really need that in the pre-commit script.

Comment (tatoMa) [reply .prettierrc:6]: > And prettier please. We really need that in the pre-commit script.

If I'm not wrong. The command "format" is the one for prettier and it's already there.

Comment (Jzeva) [reply .prettierrc:6]: Yes, I just notice that and I have deleted the comment.

**Comment by Jzeva** [comment]
image
When I tried to test the commit process I met a problem which is shown in the image above. It said that there is a syntax error in the node_modules which I never modified. Did I do anything wrong?

**Comment by tatoMa** [comment]
> image
> 
> When I tried to test the commit process I met a problem which is shown in the image above. It said that there is a syntax error in the node_modules which I never modified. Did I do anything wrong?

If Im not wrong. This kind of error is normally related to the ES version in setup files.

**Comment by tatoMa** [comment]
> image
> 
> When I tried to test the commit process I met a problem which is shown in the image above. It said that there is a syntax error in the node_modules which I never modified. Did I do anything wrong?

I tried myself but haven't encountered this issue. Which command did you run?

**Comment by Jzeva** [comment]
All good, I found maybe that because I am doing the project for my company and  I forgot to switch  node version from 12 to 16. That's my problem lol. 
BTW, I have input some wrong format commit message intentially to test it and it showed that the message cannot pass the test. Great.

**Review by Jzeva — APPROVED** [review]

**Review by kritomomo — APPROVED** [review]

**Pull request merged**